mantidproject/mantid
# MatchPeaks

MatchPeaks.py is a new Python algorithm for the category Transforms.
It is required by the IndirectILLReduction workflow algorithm for the ILL backscattering IN16b.
More history of the algorithm can be found here: https://github.com/mantidproject/mantid/issues/16753.
This algorithm should work for MatrixWorkspaces that contain maximum one elastic peak.
Hidden from the user is a tolerance value that helps evaluating optimization results from FindEPP (https://github.com/mantidproject/mantid/blob/9cd427dfa5a75e38597e10df19435c65a875a56e/Framework/PythonInterface/plugins/algorithms/FindEPP.py).
This way, the algorithm is easier to use (black box behavior) and follows the example FindEPP, where only one input and one output are required and some internal variables are as well hidden from the user. 
Implemented but uncommented is an optional second input ITableWorkspace that can be a user-defined table or a table created by FindEPP for example.
Pylint complains about too-many-branches, however, I think the code is still easy to read.

**To test:**

Run Algorithm IndirectILLReduction using the automatic GUI (ILL, instrument IN16B) in order to get left and right workspaces, then run MatchPeaks to center peaks of the left or right workspace, then visualize the workspace and verify that peaks are centered now.

UnitTest data can be used (ILLIN16B_034745, ILLIN16B_127500)

Fixes #17367



---
#### Reviewer

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):
##### Code Review
##### Functional Tests
- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.
